src/eliottree/tree_format/_text.py

```diff
@@ -100,14 +100,32 @@ def _format_tree(node, format_node, get_children, options, prefix=u'', depth=0):
                                    depth=depth + 1):
             yield result
 
+def chunked_format_tree(node, format_node, get_children, options=None, chunk_size=0)->Generator[str]:
+    lines = itertools.chain(
+        [format_node(node)],
+        _format_tree(node, format_node, get_children, options or Options()),
+        [u""],
+    )
+    if chunk_size is None or chunk_size<=0:
+        yield u"\n".join(lines)
+    else:
+        write_lines = u"\n".join((line for _, line in zip(range(chunk_size), lines)))
+        while True:
+            if not write_lines:
+                break
+            yield write_lines
+            write_lines = u"\n".join(
+                (line for _, line in zip(range(chunk_size), lines))
+            )
```

Comment:
You can avoid duplicating the `write_lines = …` code by slightly restructuring your loop:

```python
while True:
  write_lines = …
  if not write_lines:
    break
  yield write_lines
```